docarray/typing/tensor/image/image_tensor.py

```diff
@@ -1,23 +1,130 @@
-from typing import Union
+from typing import TYPE_CHECKING, Any, Type, TypeVar, Union, cast
 
+import numpy as np
+
+from docarray.typing.tensor.image.abstract_image_tensor import AbstractImageTensor
 from docarray.typing.tensor.image.image_ndarray import ImageNdArray
 from docarray.utils._internal.misc import is_tf_available, is_torch_available
 
 torch_available = is_torch_available()
 if torch_available:
-    from docarray.typing.tensor.image.image_torch_tensor import ImageTorchTensor
+    import torch
 
+    from docarray.typing.tensor.image.image_torch_tensor import ImageTorchTensor
+    from docarray.typing.tensor.torch_tensor import TorchTensor
 
 tf_available = is_tf_available()
 if tf_available:
+    import tensorflow as tf  # type: ignore
+
     from docarray.typing.tensor.image.image_tensorflow_tensor import (
-        ImageTensorFlowTensor as ImageTFTensor,
+        ImageTensorFlowTensor,
     )
+    from docarray.typing.tensor.tensorflow_tensor import TensorFlowTensor
+
+
+if TYPE_CHECKING:
+    from pydantic import BaseConfig
+    from pydantic.fields import ModelField
+
+
+T = TypeVar("T", bound="ImageTensor")
+
+
+class ImageTensor(AbstractImageTensor):
```

Comment:
I think all of these generic ImageTensor, AudioTensor can be implemented without repeating every method on them.
Is it possible to just do `class ImageTensor(AnyTensor, AbstractImageTensor)` with an (almost) empty class body?
If not, then I think we should change the inheritance structure of these tensors a bit to enable this. We should not have this copy/pasted code if we can avoid it.